src/server/server_family.cc

```diff
@@ -2031,20 +2113,23 @@ void ServerFamily::ReplicaOf(CmdArgList args, ConnectionContext* cntx) {
     return;
   }
 
-  // Flushing all the data after we marked this instance as replica.
-  Transaction* transaction = cntx->transaction;
-  transaction->Schedule();
-
-  auto cb = [](Transaction* t, EngineShard* shard) {
-    shard->db_slice().FlushDb(DbSlice::kDbAll);
-    return OpStatus::OK;
-  };
-  transaction->Execute(std::move(cb), true);
+  if (should_flush_db == ShouldFlushDb::kFlush)
+    FlushEntireDb(cntx->transaction);
```

Comment:
Wouldn't this be a problem, in that after this function finishes the transaction will be concluded afterwards? Or is it ok to `Schedule()` an already concluded transaction?

Reply:
that I am not too sure about, but I'm not sure I understand your concern? The logic of `FlushEntireDb` schedules the transaction first and then executes it, which is fine.

Are you worried that replication continues past the already-executed transaction?

Maybe @romange could chime in, I saw that he originally wrote this code 🙂

Reply:
I'm worried that transactions can't be used more than once, in which case maybe something unintended can happen after we flush the DB, when we try to use the tx again